app/controllers/projects_controller.rb

```diff
@@ -1,50 +1,36 @@
 class ProjectsController < ApplicationController
-  before_action :authorize_request,
-                except: [:all_submissions, :recent_submissions]
-  before_action :find_lesson, except: [:update, :destroy]
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found_error
+
+  before_action :authenticate_request, except: [:all_submissions]
+  before_action :find_lesson
   before_action :find_project, only: [:update, :destroy]
+
   authorize_resource only: [:update, :destroy]
 
   def create
     @project = new_project
-    if @project.save
-      render_project_json(201)
-    else
-      render_errors_json
-    end
+    @project.save
+    set_recent_submissions
   end
 
   def update
-    if @project.update(project_params)
-      render_project_json(200)
-    else
-      render_errors_json
-    end
+    @project.update(project_params)
   end
 
   def destroy
     @project.destroy
-    head :ok
+    set_recent_submissions
   end
 
   def all_submissions
```

Comment:
I think we should follow restfull conventions with this controller action by renaming it `index` instead. Its nested within a lesson so it would make sense to display all submissions for that lesson with an index.